symbulate/random_variables.py

```diff
@@ -73,6 +73,29 @@ def sim(self, n):
 
         return RVResults(self.draw() for _ in range(n))
 
+    def __call__(self, input):
+        print(
+            "Warning: Calling an RV as a function simply applies the function defining\n"
+            + "the RV to the input, regardless of whether the input is a valid outcome in \n"
+            + "the probability space on which the RV is defined."
+        )
+        temp = self.probSpace.draw()
+        if isinstance(input, tuple):
+            if len(input) != len(tuple(temp)):
+                raise Exception("Input has wrong length")
+            if all(isinstance(x, type(temp[0])) for x in input):
```

Comment:
I don't think this is the right way to do it. I think you want to check that x[i] is the same type as temp[i] for all i. You are making a strong assumption that the elements of temp are all of the same type, but that may not be true. (For example, an outcome might easily be (4, "hearts") when drawing from a deck of cards.)